app/src/main/res/layout/fragment_scan_qr_code.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <FrameLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <me.dm7.barcodescanner.zxing.ZXingScannerView
+            android:id="@+id/view_scanner"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_gravity="center">
+
+        </me.dm7.barcodescanner.zxing.ZXingScannerView>
+
+        <ImageButton
+            android:layout_width="@dimen/value_100dp"
+            android:layout_height="@dimen/value_100dp"
+            android:background="@null"
+            android:id="@+id/btn_flash_on"
+            android:layout_gravity="bottom|center"
+            app:srcCompat="@drawable/ic_flash_on"/>
+
+        <ImageButton
```

Comment:
Instead of adding two ImageButton making them visible and gone and adding two different methods, isn't it better if there are one ImageButton view and one method? You can use a boolean with an initial value of true to change the image and turning on or off the flash according to its value.

Reply:
> Instead of adding two ImageButton making them visible and gone and adding two different methods, isn't it better if there are one ImageButton view and one method? You can use a boolean with an initial value of true to change the image and turning on or off the flash according to its value.

I have updated the PR, thanks for the suggestion

Reply:
Now it looks greate, Thanks for updating.